core/shared/src/main/scala/fs2/internal/Algebra.scala

```diff
@@ -239,11 +246,7 @@ private[fs2] object Algebra {
         val (spawns, resources) = s.monitor.synchronized(s.spawns.map(_._2).toList -> s.resources.map(_._2).toList)
         spawns.map(down).foldLeft(Catenable.empty: Catenable[Resource[F]])(_ ++ _) ++ Catenable.fromSeq(resources)
       }
-      def root(s: Scope[F]): Scope[F] = s.parent match {
-        case Some(p) => root(p)
-        case None => s
-      }
-      val all = down(root(this)).toList
+      val all = down(root).toList
```

Comment:
I am really worried by this. Essentially what worries me;
     - is that we are taking all resources from the stream at time when export resources was called
     - are we 100% sure there cannot be midAcquires at that time, 
     - or worse that some of that resources did not already begun its finalization due the monitor lock being placed only on #246 line?

Reply:
We discussed some of this on Gitter. It's OK if there are mid acquires b/c if there are, that means there's an unconsAsync executing in the stream, which means the mid-acquired resources cannot be visible to the current step of execution.

If a resource has already begun finalization, when `increment` is called, it will no-op.